### PR TITLE
baseValue in item info & base stat filters

### LINKED ISSUE #4541: Exclude armor 2.0 mods in `stat:` searches, and in armor stat display.
Would be nice if stay display, totals, and search separated out the armor base from the +10 a mod gives.

Justification: If I'm looking for godrolls it's important to be able to find stat:total:>60 without the influence of mods. ie. 61 is a great roll, 51+10 isn't.

Separation in display would be helpful as well (could be similar to how masterwork effect is shown?).

### PR DESCRIPTION
**Pull request by sundevour**
this will definitely be a bake next week thing, but here are:

`basestat` to accompany `stat`, ignoring plugs which change stats

`maxbasestatvalue` to accompany `maxstatvalue`, for clearer mod-free comparison of armor stats

a `baseValue` in `D2Stat` so more features can be developed with easy access to mod-less stats.
hopefully a renaissance in Y3 item comparison

no crashes or anything weird in the loadout builder, loadout optimizer, collections, etc

note the different results for this chest piece:
image
image

closes #4541

### CODE REVIEW

**Comment by sundevour** [comment]
closes #4541

File: src/app/inventory/item-types.ts [hunk 1]
```diff
@@ -300,6 +302,11 @@ export interface D1Stat extends DimStat {
   };
 }
 
+export interface D2Stat extends DimStat {
+  /** Base stat without bonus perks applied. Important for armor. */
+  baseValue: number;
```
Comment (bhollis) [inline src/app/inventory/item-types.ts]: `D1Stat` also has a `base` property - might as well just pull that up to `DimItem` and not have to worry about changing types everywhere.

File: src/app/inventory/store/stats.ts [hunk 2]
```diff
@@ -292,14 +306,31 @@ function enhanceStatsWithPlugs(
   return stats;
 }
 
+function buildBaseStats(stats: D2Stat[], sockets: DimSocket[]) {
```
Comment (bhollis) [inline src/app/inventory/store/stats.ts]: Add a doc comment?

File: src/app/inventory/store/stats.ts [hunk 3]
```diff
@@ -292,14 +306,31 @@ function enhanceStatsWithPlugs(
...
+  for (const socket of sockets) {
+    if (socket.plug && socket.plug.plugItem.investmentStats) {
+      for (const perkStat of socket.plug.plugItem.investmentStats) {
```
Comment (bhollis) [inline src/app/inventory/store/stats.ts]: Probably worth at least mentioning in a comment that this relies on armor stats never being interpolated.

File: src/app/inventory/item-types.ts (hunk 1 above)
Comment (sundevour) [reply src/app/inventory/item-types.ts]: no problem, i just do not know enough about D1 that i could be sure doing this wouldn't interfere somehow. i will swap it out.

File: src/app/inventory/store/stats.ts (hunk 3 above)
Comment (sundevour) [reply src/app/inventory/store/stats.ts]: fair

**Comment by sundevour** [comment]
i've wanted to do the same to filters since i first ran into the size of this file

**Pull request merged**